experiment/simulation/js/layout.js

```diff
@@ -0,0 +1,285 @@
+import { deleteElement } from "./gate.js";
+import { connectGate, connectRSFF, connectJKFF, unbindEvent, initRSFlipFlop, initDFlipFlop, initJKFlipFlop, refreshWorkingArea, initTFlipFlop } from "./main.js";
+import { deleteFF } from "./flipflop.js";
+
+'use strict';
+
+// Wires
+export const wireColours = ["#ff0000", "#00ff00", "#0000ff", "#bf6be3", "#ff00ff", "#00ffff", "#ff8000", "#00ff80", "#80ff00", "#ff0080", "#8080ff", "#c0c0c0"];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+
+export const setPosition = ({ top, left }) => {
+  menu.style.left = `${left}px`;
+  menu.style.top = `${top}px`;
+  menu.style.display = "block";
+};
+
+window.addEventListener("click", e => {
+  menu.style.display = "none";
+  window.selectedComponent = null;
+  window.componentType = null;
+});
+
+menuOption.addEventListener("click", e => {
+  if (e.target.innerHTML === "Delete") {
+    if (window.componentType === "gate") {
+      deleteElement(window.selectedComponent);
+    }
+    else if (window.componentType === "flipflop") {
+      deleteFF(window.selectedComponent);
+    }
+  }
+  window.selectedComponent = null;
+  window.componentType = null;
+});
+
+// Tabs
+
+function changeTabs(e) {
+  const task = e.target.parentNode.id;
+  if (window.currentTab === task) {
+    return;
+  }
+
+  if (window.currentTab != null) {
+    document.getElementById(window.currentTab).classList.remove("is-active");
+  }
+  window.currentTab = task;
+  document.getElementById(task).classList.add("is-active");
+
+  // Half adder
+  if (task === "task1") {

```

Comment:
Please use switch-case for better readability

Reply:
Done